pkg/plugin/plugin.go

```diff
@@ -108,15 +108,22 @@ type Plugin struct {
 // - If both OS and Arch match the current platform, search will stop and the command will be prepared for execution
 // - If OS matches and there is no more specific match, the command will be prepared for execution
 // - If no OS/Arch match is found, return nil
-func getPlatformCommand(cmds []PlatformCommand) []string {
+func getPlatformCommand(cmds []PlatformCommand, env map[string]string) []string {
+	getenv := func(name string) string {
```

Comment:
I am not entirely understanding what value the environment map adds to these functions if we are already setting them with `os.Setenv` in `SetupPluginEnv`. I like that this allows for the ability to pass an arbitrary map of overrides that fallback on the environment variables, but am not sure if that is how they need to be used. Am I missing something here?

Reply:
It is to move away from depending on the users environment.  Depending on os envvars makes testing difficult by polluting the environment with overrides.

Reply:
OK, that sounds like a good enough reason to me 🙂